### PR TITLE
Support Terraform 0.12

### PR DESCRIPTION
**Pull request by jonahb**
Converts the plugin to Terraform 0.12. Terraform 0.12 requires Go modules, which requires an upgrade of the Akamai client library. Two incompatibilities with the new library are fixed. Finally, since Go modules can lock dependency versions, the vendor directory is removed.

Tested
- `go test ./...`
- Added provider to test Terraform project
- Created, updated, and deleted CNAME with `terraform apply`
- Verified results with Akamai CLI

### CODE REVIEW

File: akamai/resource_akamai_fastdns_zone.go [hunk 1]
```diff
@@ -1287,3 +1288,26 @@ func resourceFastDNSZoneExists(d *schema.ResourceData, meta interface{}) (bool,
 	zone, err := dns.GetZone(hostname)
 	return zone != nil, err
 }
+
+func saveZone(zone *dns.Zone) error {
+	log.Printf("[DEBUG] [Akamai FastDNS] Saving zone")
+	if zone.Zone.Soa != nil {
+		zone.Zone.Soa.Serial += 1
```
Comment (hundt) [inline akamai/resource_akamai_fastdns_zone.go:1295]: Why is this necessary now? did `zone.Save()` used to do it automatically?

File: akamai/resource_akamai_fastdns_zone.go [hunk 2]
```diff
@@ -1287,3 +1288,26 @@ func resourceFastDNSZoneExists(d *schema.ResourceData, meta interface{}) (bool,
...
+	}
+	return zone.Save()
+}
+
+// 'serial' and 'minimum' were changed from int to uint in v0.5.2 of the Akamai
+// client. Terraform schema doesn't support uint, so we use these functions to
+// convert during marshaling. The program will panic on overflow. This is
+// unlikely in practice because 'minimum' tends to be low, and 'serial' starts
+// as a UNIX time (much lower than max int) and increases by one when saved.
+func convertSoaUintsToInts(soa map[string]interface{}) {
```
Comment (hundt) [inline akamai/resource_akamai_fastdns_zone.go:1305]: if I recall correctly this was an issue in more record types than just SOA.  See https://github.com/akamai/terraform-provider-akamai/issues/34

Sadly the test coverage in this plugin is very poor. If we are going to change the underlying API it might be worth writing more extensive tests if possible.

File: akamai/resource_akamai_fastdns_zone.go (hunk 1 above)
Comment (jonahb) [reply akamai/resource_akamai_fastdns_zone.go:1295]: Yes, see commit message in e03fc87372cfd4494d332b759b6cb933942144cf. Newer versions only increment the serial number if it hasn't changed, but the field containing the original serial number is not included in the map produced by `ToMap`. So, if the SOA is converted to and from a map, it appears that the serial number _has_ changed (`originalSerial` is now a zero value), and it's not incremented.

File: akamai/resource_akamai_fastdns_zone.go (hunk 2 above)
Comment (jonahb) [reply akamai/resource_akamai_fastdns_zone.go:1305]: Yes, I see int to uint changes in SRV and NAPTR as well. Wasn't sure if these were relevant to our scenarios.

I wonder about investing time in writing tests for this library. It might be a better use of time to write our own, or we could just remove Akamai from Terraform. We can discuss out of band.

**Comment by hundt** [comment]
This PR is obsolete now, right?

**Comment by jonahb** [comment]
Yes, we decided to create a different provider that supports 0.12
